### PR TITLE
Added a support for a global proxy to pip module

### PR DESCRIPTION
**Pull request by Oloremo**
### What does this PR do?
Added support for a global Salt proxy to pip module

### What issues does this PR fix or reference?
#55593

### Previous Behavior
pip module and state doesn't support salt global proxy

### New Behavior
pip module and state does support salt global proxy

### Tests written?

Yes

### Commits signed with GPG?

Yes

### CODE REVIEW

File: salt/modules/pip.py [hunk 1]
```diff
@@ -995,6 +1012,11 @@ def uninstall(pkgs=None,
         behind an authenticated proxy.  If you provide
         ``user@proxy.server:port`` then you will be prompted for a password.
 
+        .. note::
+            If the the Minion has a globaly configured proxy - it will be used
+            even if no proxy was set here. To explicitly disable proxy for pip
+            you should pass the ``False`` as a value.
```
Comment (s0undt3ch) [inline salt/modules/pip.py]: 'you should pass ``False`` as'...

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: salt/modules/pip.py (hunk 1 above)
Comment (Oloremo) [reply salt/modules/pip.py]: tx. done

**Review by s0undt3ch — COMMENTED** [review]
I would also do something like:
```python
    elif proxy is not False and config.get('proxy_host') and config.get('proxy_port'):
        if config.get('proxy_username') and config.get('proxy_password'):
            http_proxy_url = 'http://{proxy_username}:{proxy_password}@{proxy_host}:{proxy_port}'.format(**config)
        else:
            http_proxy_url = 'http://{proxy_host}:{proxy_port}'.format(**config)
```

**Comment by Oloremo** [comment]
@s0undt3ch  Very cool, added!

**Review by s0undt3ch — APPROVED** [review]

**Comment by Oloremo** [comment]
re-run full  opensuse15-py2

**Comment by Oloremo** [comment]
re-run full ubuntu1604-py2

**Comment by Oloremo** [comment]
re-run full macosxmojave-py2

**Comment by Oloremo** [comment]
re-run full windows2019-py3

**Comment by max-arnold** [comment]
Nitpick: it is worth documenting that the `no_proxy` global setting has no effect on pip

**Pull request merged**